.github/workflows/maven.yml

```diff
@@ -0,0 +1,211 @@
+
+      - name: Set up JDK ${{ env.RELEASE_JAVA_VERSION }}
+        id: install_java
+        uses: actions/setup-java@v4
+        with:
+          distribution: zulu
+          java-version: ${{ env.RELEASE_JAVA_VERSION }}
+
+      - name: Get JNI headers
+        uses: actions/download-artifact@v4
+        with:
+          name: javah
+          path: target/native
+
+      - name: Install autotools
+        run: brew install autoconf automake libtool
+
+      - name: Build natives
+        run: resources/macos-build.sh "${{ steps.install_java.outputs.path }}" "${{ matrix.arch }}" "${{ github.workspace }}"
+
+      - name: Upload lib as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: darwin-${{ matrix.arch }}
+          path: src/main/resources/*
+
+  release:
+    name: Test including natives and release
+    needs:
+      - javah
+      - ubuntu_natives
+      - mac_natives
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Set up JDK ${{ env.RELEASE_JAVA_VERSION }}
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: ${{ env.RELEASE_JAVA_VERSION }}
+          cache: maven
+
+      - name: Download natives
+        uses: actions/download-artifact@v4
+        with:
+          path: target
+
+      - name: Copy natives for Maven
+        run: |
+          BASEDIR=$(pwd)/src/main/resources
+          mkdir -p $BASEDIR
+          cd target
+          for dist in */*/ ; do
+            last_dir=$(basename $dist)
+            if [[ "$last_dir" =~ ^(linux|darwin) ]]; then
+              mkdir -p "$BASEDIR/$last_dir" || true
+              cp "$dist"/*.{so,dylib} "$BASEDIR/$last_dir" || true
+            fi;
+          done;
+
+      - name: Set version
+        run: |
+          VERSION=`git describe --match "v[0-9\.]*" --long --dirty --always`
+          mvn -B versions:set -DnewVersion=${VERSION:1} -DgenerateBackupPoms=false
+          echo "VERSION=$VERSION" >> "$GITHUB_ENV"
+
+      - name: Run Maven tests with native (if PR)
+        if: github.ref != 'refs/heads/master'
+        run: mvn -B verify -DperformRelease=true
+
+      - name: Run maven tests with native (if release)
+        if: github.ref == 'refs/heads/master'
+        run: mvn -B verify
+
+      - name: Upload coverage report
+        uses: codecov/codecov-action@v4
+
+      - name: Attach final jars (if PR)
+        uses: actions/upload-artifact@v4
+        if: github.ref != 'refs/heads/master'
+        with:
+          name: jars
+          path: target/*.jar
+
+      - name: Check out Jitsi Maven repository (if release)
+        if: github.ref == 'refs/heads/master'
+        uses: actions/checkout@v4
+        with:
+          repository: jitsi/jitsi-maven-repository
+          path: jitsi-maven-repository
+
+      - name: Release to Jitsi Maven repository (if release)
```

Comment:
Why not deploy to Central instead?

Reply:
IIRC we currently have this and other deps in our maven repo. I do agree that using Maven central is a good direction. Probably after this though, since as it stands we are just automating the build that currently gets done manually...

Reply:
Well, yes, it's automating the build - but that now includes cloning the repo. This wasn't the case on the Jenkins build, where the repo was pre-cloned. I realize this is "only" the last commit, but it's anyway huge and can start to cause conflicts with the Jenkins builds.

Reply:
We'd definitely want to disable the Jenkins build once this is enabled.  Other than that I wanted to keep our current layout and we can look at deploying to central later -- right now we just have Jitsi Desktop's dependencies in Central, and everything else is in our own repo.

Reply:
The split between Central and the GitHub repo is IMO more along the lines of: nobody has taken care of writing the necessary build jobs.

Reply:
Hm, looking at https://github.com/jitsi/jitsi-sctp/pull/3 I see it pointed out that GitHub runners only have 14 Gb of storage, and jitsi-maven-repo is bigger than that even when checkout out shallow.  I may need to deploy this to Central after all.

Reply:
You could try a sparse checkout. But I'm not sure how well that works with the Maven structure. Setting up Central deployment is likely the same or even less effort.